src/addons/filter/multilang2/services/handlers/multilang2.ts

```diff
@@ -0,0 +1,114 @@
+// (C) Copyright 2015 Moodle Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Injectable } from '@angular/core';
+
+import { CoreLang } from '@services/lang';
+import { CoreFilterDefaultHandler } from '@features/filter/services/handlers/default-filter';
+import { makeSingleton } from '@singletons';
+
+/**
+ * Handler to support the Multilang filter.
+ */
+@Injectable({ providedIn: 'root' })
+export class AddonFilterMultilang2HandlerService extends CoreFilterDefaultHandler {
+
+    name = 'AddonFilterMultilang2Handler';
+    filterName = 'multilang2';
+    replacementDone = false;
+
+    /**
+     * Cache of parent language(s) of a given language.
+     */
+    protected parentCache: string[] = [];
+
+    /**
+     * This function filters the received text based on the language
+     * tags embedded in the text, and the current user language or
+     * 'other', if present.
+     *
+     * @param text The text to filter.
+     * @returns string The filtered text for this multilang block.
+     */
+    async filter(text: string): Promise<string> {
+        const currentLang = await CoreLang.getCurrentLanguage();
+
+        if (text.indexOf('mlang') === -1) {
+            return text;
+        }
+
+        this.replacementDone = false;
+
+        if (!this.parentCache[currentLang] && CoreLang.getParentLanguage(currentLang)) {
+            const parentlangs = CoreLang.getParentLanguage(currentLang);
+            this.parentCache[currentLang] = parentlangs;
+        }
+
+        const search = /{\s*mlang\s+((?:[a-z0-9_-]+)(?:\s*,\s*[a-z0-9_-]+\s*)*)\s*}(.*?){\s*mlang\s*}/g;
+        const result = text.replace(
+            search,
+            (substring, language, content) => this.replaceLangs(currentLang, [substring, language, content]),
```

Comment:
```suggestion
            (subString, language, content) => this.replaceLangs(currentLang, [subString, language, content]),
```

Changed the name of a variable to make it camel case.